AppCenter/AppCenterTests/MSUtilityTests.m

```diff
@@ -1021,18 +1021,20 @@ - (void)testDispatchSyncWithTimeoutDoesNotWait {
   XCTestExpectation *expectation = [self expectationWithDescription:@"block not called."];
   [expectation setInverted:YES];
   dispatch_queue_t serialQueue = dispatch_queue_create("test", DISPATCH_QUEUE_SERIAL);
-
+  __block dispatch_semaphore_t delayedSemaphore = dispatch_semaphore_create(0);
+    
   // When
   [MSDispatcherUtil dispatchSyncWithTimeout:blockTimeout
                                     onQueue:serialQueue
                                   withBlock:^{
-                                    [NSThread sleepForTimeInterval:blockTimeout * 2];
+                                    dispatch_semaphore_wait(delayedSemaphore, dispatch_time(DISPATCH_TIME_NOW, (int64_t)(10 * NSEC_PER_SEC)));
                                     [expectation fulfill];
                                   }];
 
   // Then
   [self waitForExpectationsWithTimeout:0
                                handler:^(NSError *error) {
+                                 dispatch_semaphore_signal(delayedSemaphore);
```

Comment:
Wait. Signal here does not make any sense

Reply:
I mean there is no real checks in this test + `fulfill` (which requires to run `waitForExpectationsWithTimeout` without timeout) happens always after `dispatch_semaphore_wait`.